telemetry/definitions/commonDefinitions.json

```diff
@@ -2609,50 +2594,80 @@
+        },
+        {
+            "name": "awsq_codeGenerationIterations",
+            "description": "Measure the conversational depth in Code gen, how often does a user invoke the weaverbird agent to re-engage in Code gen",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqCodeGenerationIterationsCount" }]
+        },
+        {
+            "name": "awsq_generateCodeClick",
+            "description": "Number of times user has clicked on Generate Code after they have iterated through the approach phase",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqGenerateCodeClickCount" }]
+        },
+        {
+            "name": "awsq_approachIteration",
+            "description": "Approach iteration count",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqApproachIterationCount" }]
+        },
+        {
+            "name": "awsq_isRegeneratedCode",
+            "description": "States if user has re-generated code as a part of code generation portion",
+            "metadata": [{ "type": "enabled" }]
+        },
+        {
+            "name": "awsq_filesReviewed",
+            "description": "Number of files reviewed by user after changes are made by WB",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqFilesReviewedCount" }] 
+        },
+        {
+            "name": "awsq_isAcceptedCodeChanges",
+            "description": "States if user applied the code changes generated for the task",
+            "metadata": [{ "type": "enabled" }] 
+        },
+        {
+            "name": "awsq_repositorySize",
+            "description": "The size of the repository",
+            "metadata": [{ "type": "awsqRepositorySize" }] 
+        },
+        {
+            "name": "awsq_filesChanged",
+            "description": "Number of files that are changed as a part of the task",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqFilesChangedCount" }] 
+        },
+        {
+            "name": "awsq_thumbsUpCount",
+            "description": "Number of users users click on the thumbs up button, to mention that they are satisfied",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqThumbsUpCount" }] 
+        },
+        {
+            "name": "awsq_thumbsDownCount",
+            "description": "Number of users click on the thumbs down button to say that they are unsatisfied",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqThumbsDownCount" }]
+        },
+        {
+            "name": "awsq_isApproachAccepted",
+            "description": "User has accepted the approach generated by Weaverbird",
+            "metadata": [{ "type": "enabled" }] 
+        },
+        {
+            "name": "awsq_serviceErrors",
+            "description": "Number of errors from engineering and service availability?",
+            "unit": "Count",
+            "metadata": [{ "type": "awsqServiceErrorsCount" }]  
```

Comment:
This should _not_ be needed. If it is, then there's something wrong with the application logic. Failed metrics in the application (Toolkit) should emit with `Result=Failed` https://github.com/aws/aws-toolkit-common/blob/13cccc35c87a02461d1a4723247db65c59a066eb/telemetry/definitions/commonDefinitions.json#L51-L52

This is handled automatically if you structure the code to use `telemetry.run()` as documented in https://github.com/aws/aws-toolkit-vscode/blob/master/docs/telemetry.md

Reply:
removed